### ISSUE TITLE
Get the task id as a log-opt tag in non Cloudwatch logs logging driver

### ISSUE THREAD

**Issue by tomelliff** [aws/amazon-ecs-agent#1045]
### Summary
It would be useful to be able to use the task id in a logging drivers tagged output.

### Description
I'm starting to look at shipping logs to somewhere other than Cloudwatch logs (was useful to get a PoC up and running but Cloudwatch logs suffers massively from being difficult to view across streams, something that is 10x worse with containers) but I'd still like to be able to see the task id that is logging if possible.

I was planning on using a fluentd log driver and tag the logs with the service name but it strikes me that it would be near perfect to use the ECS generated container name.

Looking at the output of `docker ps` and https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/docker_task_engine.go#L639 I can see the autogenerated name for the container has the task definition family + version which is useful but then oddly has a random hex string at the end.

I can see you need a source of randomness here in case a service schedules multiple tasks on to the same container instance but it would be more useful if this was just the task id which is already unique.

The task struct already contains the `Arn` which has the task id so I'd suggest parsing it out of that.

**Comment by melo** [comment]
Enable `ECS_ENABLE_CONTAINER_METADATA` and your containers will have access to the information you want. See also [Amazon ECS Container Metadata](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html) for more details.

FYI, we are still using Cloudwatch as the logs aggregator, it helps to make the new "Logs" tabs at the task and service level in the Console UI work. But then, we collect the logs into a Kinesis stream and have an ECS service that consumes the Cloudwatch Logs events and sends them to a Logs SaaS for reporting and analysis. Works fine.

The main problem is the issues with Cloudwatch Logs downtime, you'll need to be careful configuring Docker. Search for open issues related to cloudwatch logs for more details.

**Comment by tomelliff** [comment]
Yeah I've played with that but it relies on the service itself writing that to the logging output. I am, however, using it to write the container instance ID so that I can get back to the instance it's running on (although it would be better yet if I could directly see the instance ID too).

Ideally I'd like to be able to set a tag on the log driver that contains the task ID so fluentd can set it as an attribute/field in the record it writes to Elasticsearch. Right now unless the task writes this information to the logging data it sends to fluentd there's no good way to get it so you can't link the log entry to a specific task if there are multiple tasks for the service running on the same container instance.
